macros/plugins/athena/refresh_external_tables.sql

```diff
@@ -0,0 +1,61 @@
+{% macro athena__refresh_external_table(source_node) %}
```

Comment:
(After glancing quickly over many lines of code that look vaguely familiar, like an old dream): How much overlap is there between the logic here and the Redshift macros? Would it make sense to abstract the shared logic into a set of common macros, called by both?

Reply:
A ton of overlap! I was torn between being DRY and avoiding creating a dependency between the Athena and Redshift plugins, which may not have such dependencies on the AWS side (ie. Athena and Redshift syntax could diverge). I ended up not following either option 100%. This situation reminds me of guidance to avoid shared dependencies in microservices, and my own experience writing ETLs (I re-wrote a lot ETLs to not have shared code, because data can change and transformations can change between ETLs).

I think either option is fine. If you would decide which option is preferred, I'd make the changes in the code to follow!